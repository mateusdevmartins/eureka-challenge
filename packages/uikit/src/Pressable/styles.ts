import styled from 'styled-components';

export const PressableView = styled.button`
  width: 100%;
  justify-content: space-between;
  display: flex;
  flex-direction: row;
  align-items: center;
  border-width: 0px;
  background-color: rgba(255, 255, 255, 1);
`;
