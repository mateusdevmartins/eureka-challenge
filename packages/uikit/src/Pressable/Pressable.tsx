import { ReactNode, useState } from 'react';
import { PressableView } from './styles';

export interface PressableProps {
  onPress: any;
  children: ReactNode;
}

export function Pressable({ onPress, children }: PressableProps) {
  const [touched, touchedSet] = useState(false);

  return (
    <PressableView
      data-testid="pressable"
      style={{ opacity: touched ? 0.3 : 1, transition: 'opacity 500ms ease' }}
      onMouseDown={() => touchedSet(true)}
      onMouseUp={() => touchedSet(false)}
      onClick={onPress}
    >
      {children}
    </PressableView>
  );
}
