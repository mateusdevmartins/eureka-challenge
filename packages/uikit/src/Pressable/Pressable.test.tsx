import { fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Pressable } from './Pressable';

describe('Pressable', () => {
  const props = {
    onPress: jest.fn(),
  };
  it('should match snapshot', () => {
    const component = renderer.create(
      <Pressable {...props}>
        <></>
      </Pressable>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('behavior', () => {
    const components = render(
      <Pressable {...props}>
        <></>
      </Pressable>
    );
    const { getByTestId } = components;
    expect(getByTestId('pressable')).toBeTruthy();
    fireEvent.click(getByTestId('pressable'));
    expect(props.onPress).toBeCalled();
  });
});
