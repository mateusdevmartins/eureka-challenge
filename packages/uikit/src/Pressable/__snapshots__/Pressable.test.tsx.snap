// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pressable  should match snapshot 1`] = `
<button
  className="styles__PressableView-sc-1513d5n-0 gxXVdA"
  onClick={[MockFunction]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "opacity": 1,
      "transition": "opacity 500ms ease",
    }
  }
/>
`;

exports[`Pressable should match snapshot 1`] = `
<button
  className="styles__PressableView-sc-1513d5n-0 gxXVdA"
  onClick={[MockFunction]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "opacity": 1,
      "transition": "opacity 500ms ease",
    }
  }
/>
`;
